### ISSUE TITLE
Wrong Caching Shapefiles

### ISSUE THREAD

**Issue by MMTLukas** [calvinmetcalf/shapefile-js#44]
When using your library like that

```javascript
convertSHP2GeoJSON: (buffer) ->
    shp(buffer).then (geoJSON) ->
        return geoJSON
```

the item (buffer) gets added to the cache. When using the shp function with another buffer, the already added but wrong item gets returned. Which is not the way it should work!

```javascript
function shp(base, whiteList) {
    if (cache.has(base)) {
        return Promise.resolve(cache.get(base));
    }
    return shp.getShapefile(base, whiteList).then(function (resp) {
        cache.set(base, resp);
        return resp;
    });
}
```
We think there is a problem with the `set` and `has` logic inside that functions. Any suggestions? 
Thanks for creating such a library!

**Comment by calvinmetcalf** [comment]
ok I see the issue

**Comment by calvinmetcalf** [comment]
published 3.2.0 which should fix it

**Comment by MMTLukas** [comment]
Wow! Thanks for the fast replay and fix!!! +1
